modules/client/src/main/java/org/apache/ignite/jdbc/ConnectionProperties.java

```diff
@@ -0,0 +1,444 @@
+     */
+    String getUrl();
+
+    /**
+     * Set the URL.
+     *
+     * @param url The URL of the connection.
+     * @throws SQLException On invalid URL.
+     */
+    void setUrl(String url) throws SQLException;
+
+    /**
+     * Get the addresses.
+     *
+     * @return Ignite nodes addresses.
+     */
+    HostAndPortRange[] getAddresses();
+
+    /**
+     * Set the ignite node addresses.
+     *
+     * @param addrs Ignite nodes addresses.
+     */
+    void setAddresses(HostAndPortRange[] addrs);
+
+
+    /**
+     * Get the auto close server cursors flag.
+     *
+     * @return Auto close server cursors flag.
+     */
+    boolean isAutoCloseServerCursor();
+
+    /**
+     * Set the auto close server cursors flag.
+     *
+     * @param autoCloseServerCursor Auto close server cursors flag.
+     */
+    void setAutoCloseServerCursor(boolean autoCloseServerCursor);
+
+    /**
+     * Get the socket send buffer size.
+     *
+     * @return Socket send buffer size.
+     */
+    int getSocketSendBuffer();
+
+    /**
+     * Set the socket send buffer size.
+     *
+     * @param size Socket send buffer size.
+     * @throws SQLException On error.
+     */
+    void setSocketSendBuffer(int size) throws SQLException;
+
+    /**
+     * Get the socket receive buffer size.
+     *
+     * @return Socket receive buffer size.
+     */
+    int getSocketReceiveBuffer();
+
+    /**
+     * Set the socket receive buffer size.
+     *
+     * @param size Socket receive buffer size.
+     * @throws SQLException On error.
+     */
+    void setSocketReceiveBuffer(int size) throws SQLException;
+
+    /**
+     * Get the TCP no delay flag.
+     *
+     * @return TCP no delay flag.
+     */
+    boolean isTcpNoDelay();
+
+    /**
+     * Set the TCP no delay flag.
+     *
+     * @param tcpNoDelay TCP no delay flag.
+     */
+    void setTcpNoDelay(boolean tcpNoDelay);
+
+    /**
+     * Gets SSL connection mode.
+     *
+     * @return Use SSL flag.
+     * @see #setSslMode(String)
+     */
+    String getSslMode();
+
+    /**
+     * Use SSL connection to Ignite node. In case set to {@code "require"} SSL context must be configured.
+     * {@link #setSslClientCertificateKeyStoreUrl} property and related properties must be set up
+     * or JSSE properties must be set up (see {@code javax.net.ssl.keyStore} and other {@code javax.net.ssl.*}
+     * properties)
+     *
+     * In case set to {@code "disable"} plain connection is used.
+     * Available modes: {@code "disable", "require"}. Default value is {@code "disable"}
+     *
+     * @param mode SSL mode.
+     */
+    void setSslMode(String mode);
+
+    /**
+     * Gets protocol for secure transport.
+     *
+     * @return SSL protocol name.
+     */
+    String getSslProtocol();
+
+    /**
+     * Sets protocol for secure transport. If not specified, TLS protocol will be used.
+     * Protocols implementations supplied by JSEE: SSLv3 (SSL), TLSv1 (TLS), TLSv1.1, TLSv1.2
```

Comment:
I'm sure that we should support only TLSv2 and TLSv3 because previous protocols are obsolete

Reply:
@vkulichenko WDYT?

Reply:
I think this should depends on SSL Provider, as some users may have 3-rd party one.

Reply:
I created separate task IGNITE-15428 for SSL connection. I suggest make all ssl-related changes in a separate ticket.